internal/machine/update.go

```diff
@@ -18,6 +18,82 @@ func Update(ctx context.Context, m *api.Machine, input *api.LaunchMachineInput)
 		colorize    = io.ColorScheme()
 	)
 
+	if input != nil && input.Config != nil && input.Config.Guest != nil {
+		// Check that there's a valid number of CPUs
+		var validNumCpus []int
+
+		if input.Config.Guest.CPUKind == "shared" {
+			validNumCpus = append(validNumCpus, 1, 2, 4, 8)
+
+		} else if input.Config.Guest.CPUKind == "performance" {
+			validNumCpus = append(validNumCpus, 1, 2, 4, 8, 16)
+
+		}
+
+		validCpuNum := false
+
+		for _, num := range validNumCpus {
+			if num == input.Config.Guest.CPUs {
+				validCpuNum = true
+				break
+
+			}
+		}
+
+		if !validCpuNum {
+			return fmt.Errorf("invalid config: invalid number of CPUs for %s guest. Valid numbers are %v\nView more information here: https://fly.io/docs/about/pricing/#machines", input.Config.Guest.CPUKind, validNumCpus)
+
+		}
+
+		if input.Config.Guest.CPUKind == "shared" {
+			if input.Config.Guest.MemoryMB%256 != 0 {
```

Comment:
Channeling jp here: these can be one if, not nested :)

Reply:
roger roger